pkg/kubelet/dockertools/docker.go

```diff
@@ -153,6 +154,14 @@ func filterHTTPError(err error, image string) error {
 
 // Check if the inspected image matches what we are looking for
 func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
+	// NB: the code below this check checks for validity of `image` as a pull spec.
+	// However, it can be useful to also inspect images by ID (config digest, e.g.
```

Comment:
I don't like the `checked` arg added to the function. If the verification is not always needed, why not just move it up to somewhere else? E.g., `IsImagePresent` might be a good candidate.